### PR TITLE
Fix crash when disconnecting before connecting.

### LINKED ISSUE #107: Location listener demo crashes on first launch
### Description

After installing demo app and launching for the first time if you navigate to the "LocationListener" demo activity the app crashes. User is never prompted to agree to location permissions.

### Steps to Reproduce

1. Uninstall demo app (if installed)
2. Install demo app
3. Launch and navigate to FusedLocationApi > LocationListener example
4. Runtime permission dialog will not be displayed and app will force quit

### Lost & Android Version

Lost 1.1.2-SNAPSHOT (local build on master)
Android 6.0.1

### PR DESCRIPTION
**Pull request by clydebarrow**
Fixes #107.

### CODE REVIEW

File: lost/src/main/java/com/mapzen/android/lost/internal/FusedLocationProviderApiImpl.java [hunk 1]
```diff
@@ -68,11 +70,12 @@
...
-    return connecting;
+    return connectState == ConnectState.CONNECTING;
   }
 
   public FusedLocationProviderApiImpl() {
     connectionCallbacks = new HashSet<>();
+    connectState = ConnectState.IDLE;
   }
 
   public void connect(Context context, LostApiClient.ConnectionCallbacks callbacks) {
```
Comment (sarahlensing) [inline lost/src/main/java/com/mapzen/android/lost/internal/FusedLocationProviderApiImpl.java]: When `connect` is called we should update the `connectState` to `ConnectState.CONNECTING`

**Comment by clydebarrow** [comment]
I've added a test for that specific case, and refined the handling of `connectState`, building on your suggested change. It's now possible for the state to go from IDLE to CONNECTING back to IDLE cleanly. Also fixed test failures revealed as a consequence. All tests now pass.

File: lost/src/main/java/com/mapzen/android/lost/internal/FusedLocationProviderApiImpl.java [hunk 2]
```diff
@@ -19,6 +10,15 @@
 import android.os.Looper;
 import android.util.Log;
 
+import com.mapzen.android.lost.api.FusedLocationProviderApi;
```
Comment (ecgreb) [inline lost/src/main/java/com/mapzen/android/lost/internal/FusedLocationProviderApiImpl.java]: Library imports above android imports

File: lost/src/main/java/com/mapzen/android/lost/internal/LostApiClientImpl.java [hunk 3]
```diff
@@ -18,7 +18,8 @@ public LostApiClientImpl(Context context, ConnectionCallbacks callbacks) {
     this.connectionCallbacks = callbacks;
   }
 
-  @Override public void connect() {
```
Comment (ecgreb) [inline lost/src/main/java/com/mapzen/android/lost/internal/LostApiClientImpl.java]: Inline method annotations preferred

File: lost/src/main/java/com/mapzen/android/lost/internal/LostApiClientImpl.java [hunk 4]
```diff
@@ -1,10 +1,10 @@
 package com.mapzen.android.lost.internal;
 
+import android.content.Context;
+
 import com.mapzen.android.lost.api.LocationServices;
 import com.mapzen.android.lost.api.LostApiClient;
 
-import android.content.Context;
```
Comment (ecgreb) [inline lost/src/main/java/com/mapzen/android/lost/internal/LostApiClientImpl.java]: Library imports above android imports

File: lost/src/main/java/com/mapzen/android/lost/internal/LostApiClientImpl.java [hunk 5]
```diff
@@ -28,20 +29,22 @@ public LostApiClientImpl(Context context, ConnectionCallbacks callbacks) {
...
+        fusedApi.connectionCallbacks.add(connectionCallbacks);
+      }
+    } else {
+      fusedApi.connect(context, connectionCallbacks);
     }
-
     clientManager.addClient(this);
   }
 
-  @Override public void disconnect() {
```
Comment (ecgreb) [inline lost/src/main/java/com/mapzen/android/lost/internal/LostApiClientImpl.java]: Inline method annotations preferred

File: lost/src/main/java/com/mapzen/android/lost/internal/LostApiClientImpl.java [hunk 6]
```diff
@@ -52,7 +55,8 @@ public LostApiClientImpl(Context context, ConnectionCallbacks callbacks) {
     getFusedLocationProviderApiImpl().disconnect();
   }
 
-  @Override public boolean isConnected() {
```
Comment (ecgreb) [inline lost/src/main/java/com/mapzen/android/lost/internal/LostApiClientImpl.java]: Inline method annotations preferred

File: lost/src/test/java/com/mapzen/android/lost/internal/FusedLocationProviderApiImplTest.java [hunk 7]
```diff
@@ -1,5 +1,11 @@
 package com.mapzen.android.lost.internal;
 
+import android.app.PendingIntent;
```
Comment (ecgreb) [inline lost/src/test/java/com/mapzen/android/lost/internal/FusedLocationProviderApiImplTest.java]: Library imports above android imports

**Pull request merged**